app/containers/WeatherContainer/index.js

```diff
@@ -0,0 +1,263 @@
+/**
+ *
+ * WeatherContainer
+ *
+ */
+
+import React, { useEffect, memo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl, FormattedMessage as T } from 'react-intl';
+import Clickable from '@components/Clickable';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { useInjectSaga } from '@utils/injectSaga';
+import { selectWeatherContainer, selectCityData, selectCityError, selectCityName } from './selectors';
+import { weatherContainerCreators } from './reducer';
+import saga from './saga';
+import { Card, Skeleton, Input } from 'antd';
+import styled from 'styled-components';
+import debounce from 'lodash/debounce';
+import isEmpty from 'lodash/isEmpty';
+const { Search } = Input;
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 20px 0;
+    max-width: ${props => props.maxwidth};
+    color: ${props => props.color};
+    ${props => props.color && `color: ${props.color}`};
+  }
+`;
+const Container = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    max-width: ${props => props.maxwidth}px;
+    width: 100%;
+    margin: 0 auto;
+    padding: ${props => props.padding}px;
+  }
+`;
+const RightContent = styled.div`
+  display: flex;
+  align-self: flex-end;
+`;
+export function WeatherContainer({ dispatchCityWheather, cityData = {}, cityError = null, cityName, intl }) {
+  useInjectSaga({ key: 'weatherContainer', saga });
+
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if ((loading && cityData) || cityError) {
+      setLoading(false);
+    }
+  }, [cityData]);
+
+  useEffect(() => {
+    if (cityName && !cityData) {
+      dispatchCityWheather(cityName);
+      setLoading(true);
+    }
+  }, []);
+
+  const handleOnChange = wName => {
+    if (!isEmpty(wName)) {
+      dispatchCityWheather(wName);
+      setLoading(true);
+    }
+  };
+  const debouncedHandleOnChange = debounce(handleOnChange, 400);
+
+  const renderCityList = () => {
+    return (
+      (cityData || loading) && (
+        <CustomCard>
+          <Skeleton loading={loading} active>
+            {cityName && <div></div>}
+
+            <CustomCard>
+              <h2>City Name - {cityData.name}</h2>
+              <h2>City Timezone - {cityData.timezone}</h2>
+              <h2>City Temperature - {cityData.main.temp}</h2>
+
+              {/* <h2 id="rpository_name" values={{ name: cityData.name }} />
+              <h2 id="repository_full_name" values={{ fullName: cityData.temp_min }} /> */}
+            </CustomCard>
+          </Skeleton>
+        </CustomCard>
+      )
+    );
+  };
+  const renderErrorState = () => {
+    let citywError;
+    if (cityError) {
+      citywError = cityError;
+    }
+    return (
+      !cityData &&
+      citywError && (
```

Comment:
Let's have a more meaningful variable name. 
Also, please ensure that all variables are always in camelCase